### ISSUE TITLE
Spectral concatenation

### ISSUE THREAD

**Issue by brechmos-stsci** [astropy/specutils#318]
I think there is a need for a Spectrum1D concatenate method that takes some number of Spectrum1D objects and outputs a single one. 

There are a lot of sketchy issues with that, but it could be useful for applying to the output of extract_region in order to get the one Spectrum1D.

Some issues:
- How to deal with different spectral axes units from the different components?  
  - I would suggest we convert all to the spectral_axis unit of the first Spectrum1D in the list
- How to deal with different flux units from the different components?
  - I would suggest we convert all to the flux unit of the first Spectrum1D in the list
- Do we assume the list of Spectrum1D objects passed in are in the "correct order"?  ie what do we do if the second Spectrum1D spectral axis is completely before the first?  
  - I would suggest the Spectrum1D objects be ordered based on the first spectral_axis value of each AND produce a warning to say we re-ordered
- How should we deal with overlapping Spectrum1D objects passed in? 
  - I think we should raise an Exception, for now, saying overlapping Spectrum1D objects can not be concatenated.

Based on some thoughts and discussion in https://github.com/astropy/specutils/pull/289#issuecomment-419448293

@eteq

**Comment by brechmos-stsci** [comment]
Also... I suspect this should be in specutils.manipulation.concatenate (or concat) and it can take a list of Spectrum1D objects or an arbitrary number of Spectrum1D objects as input.

**Comment by nmearl** [comment]
So something beyond the arithmetic addition operation that would create a new WCS to represent something like a masked version of what existed before the use of the `SpectrumRegion`?

**Comment by nmearl** [comment]
No, no, sorry, I misspoke. I was just clarifying about having an operation that produces something _like_ the original input spectrum, but "with holes" in the WCS where the `SpectralRegion` excised (either explicitly or by omission) that portion of the spectrum.

**Comment by nmearl** [comment]
> How to deal with different spectral axes units from the different components?
>    I would suggest we convert all to the spectral_axis unit of the first Spectrum1D in the list

This is done in the current splice documentation.l

> How to deal with different flux units from the different components?
>    I would suggest we convert all to the flux unit of the first Spectrum1D in the list

The spectrum arithmetic handles the flux unit conversions in the splice docs.

> Do we assume the list of Spectrum1D objects passed in are in the "correct order"? ie what do we do if the second Spectrum1D spectral axis is completely before the first?
>    I would suggest the Spectrum1D objects be ordered based on the first spectral_axis value of each AND produce a warning to say we re-ordered

This is not currently handled in the splice documentation which demonstrates concatenating spectral axes _in order_. However, I have not tested this locally and it may be possible that the resampling logic forces a sort on the spectral axis (or otherwise raises an error that it's not in ascending order, which is necessary for the resampling to work).

> How should we deal with overlapping Spectrum1D objects passed in?
>    I think we should raise an Exception, for now, saying overlapping Spectrum1D objects can not be concatenated.

Currently, the splice method first finds the new spectral axis that encompasses both input spectra, then resamples the input spectrum onto this spectral axis. Afterward, the spectra are added together. So overlapping portions would have their flux values added, which I _believe_ is the most obvious behavior.
